### ISSUE TITLE
partial purge facility

### ISSUE THREAD

**Issue by kohlhase** [slatex/sTeX-React#214]
ALeA is becoming more complete, and the data the system collects becomes more diverse. 

Therefore we may want to have a more granular "purge my data" functionality and interface. 

For instance we might want to only purge study buddy data while leaving the learner model intact. Or we might want to purge all interactions in 2023 and before (because I made the new year's resolution that I am becoming a completely new student :D)

I am not sure what the relevant data categories might be, so we may want to collect them here.

**Comment by lambdaTotoro** [comment]
One category that immediately springs to mind is "anything to do only with $COURSE" for when you're deciding to drop AI1 in the middle of the semester, but you don't want to erase last year's stellar IWGS performance. Or something along those lines.

One question that immediately springs to mind is what is supposed to happen to the learner model if you only partially erase it, especially in time. I'm assuming you want to recompute it as if the remaining interactions were the only ones that ever happened?

**Comment by kohlhase** [comment]
Yes, I have been thinking about how to do that as well, because this is quite natural wo want. I guess a complete recomputation with purged log files is all we can do.
